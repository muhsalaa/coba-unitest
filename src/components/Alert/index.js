import React from 'react';
import clsx from 'clsx';
import '../../styles/main.css';

function Alert({ color, children, className, icon, ...resProps }) {
  const colors = {
    green: 'bg-green-200 text-green-700',
    red: 'bg-red-200 text-red-700',
  };

  const alertClass = clsx(
    'my-2 p-3 text-lg rounded-md flex items-center justify-center',
    colors[color],
    className
  );

  return (
    <div data-testid="qa-alert" className={alertClass} {...resProps}>
      {icon && React.createElement(icon, { className: 'mr-2 text-xl' })}
      {children}
    </div>
  );
}

Alert.defaultProps = {
  color: 'green',
  icon: null,
};

export default Alert;
