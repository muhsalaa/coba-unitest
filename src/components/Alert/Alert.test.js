import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { FaExclamationTriangle } from 'react-icons/fa';

import Alert from './index';

describe('Alert component', () => {
  describe('Component Renders', () => {
    // tulis ini semua dulu baru masukkin ke describe lg
    it('Renders correctly and match snapshot', () => {
      const wrapper = mount(<Alert />);

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Renders text (children) provided', () => {
      const wrapper = shallow(<Alert>Hola como estas</Alert>);

      // jelaskan macam2 matchers
      expect(wrapper.text()).toBe('Hola como estas');
    });

    it('Renders icon if provided', () => {
      const wrapper = shallow(<Alert icon={FaExclamationTriangle}>Hola</Alert>);

      expect(wrapper.text()).toContain('<FaExclamationTriangle />');
      expect(wrapper.text()).toBe('<FaExclamationTriangle />Hola');
      expect(wrapper.text()).toEqual('<FaExclamationTriangle />Hola');
      expect(wrapper.text()).toMatch(/<FaExclamationTriangle \/>Hola/);
    });
  });

  describe('Component style', () => {
    it('Should have green color by default', () => {
      const wrapper = mount(<Alert />);
      const alert = wrapper.find('div[data-testid="qa-alert"]');

      expect(alert.props().className).toContain('bg-green-200');
    });

    it('Should have red color if props provided', () => {
      const wrapper = mount(<Alert color="red" />);
      const alert = wrapper.find('div[data-testid="qa-alert"]');

      expect(alert.props().className).toContain('bg-red-200');
    });
  });
});
