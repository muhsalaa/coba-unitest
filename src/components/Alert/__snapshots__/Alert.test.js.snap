// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component Component Renders Renders correctly and match snapshot 1`] = `
<Alert
  color="green"
  icon={null}
>
  <div
    className="my-2 p-3 text-lg rounded-md flex items-center justify-center bg-green-200 text-green-700"
    data-testid="qa-alert"
  />
</Alert>
`;
